apps/aechannel/src/aesc_utils.erl

```diff
@@ -130,67 +133,72 @@ deserialize_payload(<<>>) ->
 deserialize_payload(Payload) ->
     try
         SignedTx = aetx_sign:deserialize_from_binary(Payload),
-        Tx       = aetx_sign:tx(SignedTx),
-        case aetx:specialize_type(Tx) of
-            {channel_offchain_tx, PayloadTx} ->
+        case unpack_payload(SignedTx) of
+            {ok, PayloadTx} ->
                 {ok, SignedTx, PayloadTx};
-            _ ->
-                {error, bad_offchain_state_type}
+            Err = {error, _} ->
+                Err
         end
     catch _:_ ->
             {error, payload_deserialization_failed}
     end.
 
+unpack_payload(Tx) ->
+    case aetx:specialize_type(aetx_sign:tx(Tx)) of
+        {channel_offchain_tx, OffChainTx} -> {ok, OffChainTx};
+        {ga_meta_tx, GAMetaTx}            -> unpack_payload(aega_meta_tx:tx(GAMetaTx));
```

Comment:
🤔 Good catch I think, it will certainly be caught further on (since there is no generalized account that could have properly signed this payload) but it should be stopped earlier.